distributions/aws/test/e2e/tests/test_cognito.py

```diff
@@ -0,0 +1,53 @@
+import requests
+import pytest
+
+from e2e.utils.config import metadata, configure_env_file
+from e2e.conftest import region
+
+from e2e.fixtures.cluster import cluster
+from e2e.fixtures.kustomize import kustomize
+from e2e.fixtures.clients import account_id
+
+from e2e.fixtures.cognito_dependencies import (
+    cognito_bootstrap,
+    configure_ingress,
+    configure_kf_admin_role,
+    post_deployment_dns_update,
+)
+
+
+@pytest.fixture(scope="class")
+def kustomize_path():
+    return "../../examples/cognito"
+
+
+@pytest.fixture(scope="class")
+def configure_manifests(region, cluster, configure_ingress):
+    print(
+        "fixture to introduce dependency to setup cognito related resources before applying manifests"
+    )
+
+
+class TestCognito:
+    @pytest.fixture(scope="class")
+    def setup(self, metadata, post_deployment_dns_update):
+        metadata_file = metadata.to_file()
+        print(metadata.params)  # These needed to be logged
+        print("Created metadata file for TestSanity", metadata_file)
+
+    def test_url_is_up(self, setup, cognito_bootstrap):
+        subdomain_name = cognito_bootstrap["route53"]["subDomain"]["name"]
+        kubeflow_endpoint = "https://kubeflow." + subdomain_name
+        response = requests.get(kubeflow_endpoint)
+        assert response.status_code == 200
+        # request was redirected
+        assert len(response.history) > 0
+        # redirection was to cognito domain
+        assert "auth." + subdomain_name in response.url
```

Comment:
This is not testing login correct but just url redirection. Are we planning to test login ?

Reply:
We can test login using a browser but not using API calls right now. Browser testing can be separate PR

another way would be to use kubeconfig instead of session or cookies but I am not sure if its the right UX and we are not testing multi-user at that point.